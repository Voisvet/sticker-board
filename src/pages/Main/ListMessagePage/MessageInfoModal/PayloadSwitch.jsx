import React from 'react';

import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import CloudDownload from '@material-ui/icons/CloudDownload';
import { withStyles } from '@material-ui/core/styles';

// --------------------------------------------------
//
//  Styles section
//
// --------------------------------------------------

const styles = theme => ({
  leftIcon: {
    marginRight: theme.spacing.unit,
  },
  button: {
    width: "100%"
  },
  image: {
    marginRight: "auto",
    marginLeft: "auto",
    display: "block"
  }
});

// --------------------------------------------------
//
//  Main component section
//
// --------------------------------------------------

const PayloadSwitch = (props) => {

  // ------------------------------
  //
  //  Props extraction
  //
  // ------------------------------

  const {payload, payload_type, classes} = props;

  // ------------------------------
  //
  //  Special case - content loading
  //
  // ------------------------------

  if (payload_type !== 'file' && !payload) {
    return (
      <Typography variant='body2'>
        Payload is loading...
      </Typography>
    );
  }

  // ------------------------------
  //
  //  Render part
  //
  // ------------------------------

  console.log(payload_type);
  if (payload_type === 'file') {
    return (
      <Button variant="contained" color="primary" className={classes.button}>
        <CloudDownload className={classes.leftIcon} />
        Download File
      </Button>
    );
  } else if (payload_type === 'message') {
    return (
      <Typography variant='body2'>
        { payload }
      </Typography>
    );
  } else if (payload_type === 'sticker') {
    let src = 'data:image/png;base64,' + payload;
    return <img className={classes.image} src={src}/>;
  } else {
    return (
      <Typography variant='body2'>
        There is no payload o_O
      </Typography>
    );
  }
};

// --------------------------------------------------
//
//  Composing and export section
//
// --------------------------------------------------

export default withStyles(styles)(PayloadSwitch);
