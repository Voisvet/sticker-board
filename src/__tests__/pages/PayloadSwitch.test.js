import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import PayloadSwitch from '../../pages/Main/ListMessagePage/MessageInfoModal/PayloadSwitch';

Enzyme.configure({ adapter: new Adapter() });

describe("Test PayloadSwitch rendering...", () => {
  test("payload type = file", () => {
    const component = Enzyme.shallow(
      <PayloadSwitch payload_type="file" payload={undefined}/>
    );
    expect(component).toMatchSnapshot();
  });

  test("payload type = sticker, without actual payload", () => {
    const component = Enzyme.shallow(
      <PayloadSwitch payload_type="sticker" payload={undefined}/>
    );
    expect(component).toMatchSnapshot();
  });

  test("payload type = message, without actual payload", () => {
    const component = Enzyme.shallow(
      <PayloadSwitch payload_type="message" payload={undefined}/>
  );
    expect(component).toMatchSnapshot();
  });

  test("payload type = sticker, with payload", () => {
    const component = Enzyme.shallow(
      <PayloadSwitch payload_type="sticker" payload="base64_encoded_image"/>
    );
    expect(component).toMatchSnapshot();
  });

  test("payload type = message, with payload", () => {
    const component = Enzyme.shallow(
      <PayloadSwitch payload_type="message" payload="I'm a message!!!"/>
    );
    expect(component).toMatchSnapshot();
  });
});
