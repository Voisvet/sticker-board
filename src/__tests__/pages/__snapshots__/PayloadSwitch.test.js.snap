// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PayloadSwitch rendering... payload type = file 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PayloadSwitch)
    payload_type="file"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "PayloadSwitch-button-2",
        "image": "PayloadSwitch-image-3",
        "leftIcon": "PayloadSwitch-leftIcon-1",
      },
      "payload": undefined,
      "payload_type": "file",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "PayloadSwitch-button-2",
          "image": "PayloadSwitch-image-3",
          "leftIcon": "PayloadSwitch-leftIcon-1",
        },
        "payload": undefined,
        "payload_type": "file",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Test PayloadSwitch rendering... payload type = message, with payload 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PayloadSwitch)
    payload="I'm a message!!!"
    payload_type="message"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "PayloadSwitch-button-2",
        "image": "PayloadSwitch-image-3",
        "leftIcon": "PayloadSwitch-leftIcon-1",
      },
      "payload": "I'm a message!!!",
      "payload_type": "message",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "PayloadSwitch-button-2",
          "image": "PayloadSwitch-image-3",
          "leftIcon": "PayloadSwitch-leftIcon-1",
        },
        "payload": "I'm a message!!!",
        "payload_type": "message",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Test PayloadSwitch rendering... payload type = message, without actual payload 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PayloadSwitch)
    payload_type="message"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "PayloadSwitch-button-2",
        "image": "PayloadSwitch-image-3",
        "leftIcon": "PayloadSwitch-leftIcon-1",
      },
      "payload": undefined,
      "payload_type": "message",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "PayloadSwitch-button-2",
          "image": "PayloadSwitch-image-3",
          "leftIcon": "PayloadSwitch-leftIcon-1",
        },
        "payload": undefined,
        "payload_type": "message",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Test PayloadSwitch rendering... payload type = sticker, with payload 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PayloadSwitch)
    payload="base64_encoded_image"
    payload_type="sticker"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "PayloadSwitch-button-2",
        "image": "PayloadSwitch-image-3",
        "leftIcon": "PayloadSwitch-leftIcon-1",
      },
      "payload": "base64_encoded_image",
      "payload_type": "sticker",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "PayloadSwitch-button-2",
          "image": "PayloadSwitch-image-3",
          "leftIcon": "PayloadSwitch-leftIcon-1",
        },
        "payload": "base64_encoded_image",
        "payload_type": "sticker",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Test PayloadSwitch rendering... payload type = sticker, without actual payload 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PayloadSwitch)
    payload_type="sticker"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "PayloadSwitch-button-2",
        "image": "PayloadSwitch-image-3",
        "leftIcon": "PayloadSwitch-leftIcon-1",
      },
      "payload": undefined,
      "payload_type": "sticker",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "PayloadSwitch-button-2",
          "image": "PayloadSwitch-image-3",
          "leftIcon": "PayloadSwitch-leftIcon-1",
        },
        "payload": undefined,
        "payload_type": "sticker",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
